components/places/src/ffi.rs

```diff
@@ -113,6 +144,151 @@ impl PlacesConnection {
             )
         })
     }
+
+    /// Add an observation to the database.
+    fn apply_observation(&self, visit: VisitObservation) -> Result<()> {
+        match self.with_conn(|conn| history::apply_observation(conn, visit)) {
+            Ok(_) => Ok(()),
+            Err(e) => Err(e),
+        }
+    }
+
+    fn get_visited_urls_in_range(
+        &self,
+        start: i64,
+        end: i64,
+        include_remote: bool,
+    ) -> Result<Vec<String>> {
+        self.with_conn(|conn| {
+            history::get_visited_urls(
+                conn,
+                // Probably should allow into()...
+                types::Timestamp(start.max(0) as u64),
+                types::Timestamp(end.max(0) as u64),
+                include_remote,
+            )
+        })
+    }
+
+    fn get_visit_infos(
+        &self,
+        start_date: i64,
```

Comment:
should these be your `Timestamp` type? (Same in the function above too) It abstracts away some of the signed/unsigned ugliness into one nice spot!